rtl8822bu-dkms Does Not Build On Recent Kernels

### Is this a new report?

Yes

### System Info

Void 5.19.16_1 x86_64 GenuineIntel uptodate FFFFF

### Package(s) Affected

rtl8822bu-dkms-20201222_2

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

_No response_

### Expected behaviour

The module builds on a kernel

### Actual behaviour

The build fails with the following in `/var/lib/dkms/88x2bu/20201222/build/make.log `:
```
DKMS make.log for 88x2bu-20201222 for kernel 5.19.16_1 (x86_64)
Sun Oct 30 02:22:31 PM EDT 2022
/bin/sh: 1: bc: not found
make ARCH="x86_64" CROSS_COMPILE="" -C /lib/modules/5.19.16_1/build M="/var/lib/dkms/88x2bu/20201222/build" modules
make[1]: Entering directory '/usr/src/kernel-headers-5.19.16_1'
/bin/sh: 1: bc: not found
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_cmd.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_security.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_debug.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_io.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_ioctl_query.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_ioctl_set.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_ieee80211.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_mlme.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_mlme_ext.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_mi.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_wlan_util.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_vht.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_pwrctrl.o
/var/lib/dkms/88x2bu/20201222/build/core/rtw_mlme_ext.c: In function 'mgt_dispatcher':
/var/lib/dkms/88x2bu/20201222/build/core/rtw_mlme_ext.c:1828:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
 1828 |    ptable->func = &OnAuthClient;
      |    ~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/var/lib/dkms/88x2bu/20201222/build/core/rtw_mlme_ext.c:1831:2: note: here
 1831 |  case WIFI_ASSOCREQ:
      |  ^~~~
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_rf.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_recv.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_sta_mgt.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_ap.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/mesh/rtw_mesh.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_xmit.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_p2p.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_rson.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_tdls.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_br_ext.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_iol.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_sreset.o
  CC [M]  /var/lib/dkms/88x2bu/20201222/build/core/rtw_btcoex_wifionly.o
/var/lib/dkms/88x2bu/20201222/build/core/rtw_br_ext.c:20:11: fatal error: net/ipx.h: No such file or directory
   20 |  #include <net/ipx.h>
      |           ^~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:249: /var/lib/dkms/88x2bu/20201222/build/core/rtw_br_ext.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:1846: /var/lib/dkms/88x2bu/20201222/build] Error 2
make[1]: Leaving directory '/usr/src/kernel-headers-5.19.16_1'
make: *** [Makefile:2013: modules] Error 2
```

### Steps to reproduce

1. Install `rtl8822bu-dkms` from the repos
2. Attempt to build module via DKMS (`xbps` trigger or otherwise)

--- a/srcpkgs/rtl8822bu-dkms/patches/0001-void-arch-generic-plumbing-bits.patch
+++ b/srcpkgs/rtl8822bu-dkms/patches/0001-void-arch-generic-plumbing-bits.patch
@@ -11,17 +11,17 @@ diff --git Makefile Makefile
 index 29da0bf..d58bc95 100644
 --- a/Makefile
 +++ b/Makefile
-@@ -97,7 +97,8 @@ CONFIG_RTW_SDIO_PM_KEEP_POWER = y
+@@ -120,7 +120,8 @@ CONFIG_RTW_SDIO_PM_KEEP_POWER = y
  ###################### MP HW TX MODE FOR VHT #######################
  CONFIG_MP_VHT_HW_TX_MODE = n
  ###################### Platform Related #######################
 -CONFIG_PLATFORM_I386_PC = y
 +CONFIG_PLATFORM_VOID_NATIVE = y
 +CONFIG_PLATFORM_I386_PC = n
- CONFIG_PLATFORM_ARM_RPI = n
  CONFIG_PLATFORM_ANDROID_X86 = n
  CONFIG_PLATFORM_ANDROID_INTEL_X86 = n
-@@ -1030,6 +1031,17 @@ endif
+ CONFIG_PLATFORM_JB_X86 = n
+@@ -1289,6 +1289,17 @@ endif
  
  EXTRA_CFLAGS += -DDM_ODM_SUPPORT_TYPE=0x04
  
@@ -36,9 +36,9 @@ index 29da0bf..d58bc95 100644
 +STAGINGMODDIR := /usr/lib/modules/$(KVER)/kernel/drivers/staging
 +endif
 +
- ifeq ($(CONFIG_PLATFORM_I386_PC), y)
- EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
- EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
+ ifeq ($(CONFIG_RTW_VHT_2G4), y)
+ EXTRA_CFLAGS += -DRTW_VHT_2G4=1
+ else
 -- 
 2.29.2
 

--- a/srcpkgs/rtl8822bu-dkms/template
+++ b/srcpkgs/rtl8822bu-dkms/template
@@ -1,16 +1,16 @@
 # Template file for 'rtl8822bu-dkms'
 pkgname=rtl8822bu-dkms
-version=20201222
-revision=2
-_gitrev=fcfd4ecca1512d4cd2db4aa91679576d2a5ab8eb
-wrksrc="rtl8822bu-${_gitrev}"
-depends="dkms"
+version=20221030
+revision=1
+_gitrev=f77c4e55cc1852b9f6e5aec416a34abcdabcd825
+wrksrc="rtl88x2bu-${_gitrev}"
+depends="dkms bc"
 short_desc="Realtek 8822BU USB WiFi driver (DKMS)"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-only"
 homepage="https://www.tp-link.com"
-distfiles="https://github.com/EntropicEffect/rtl8822bu/archive/${_gitrev}.tar.gz"
-checksum=fb2dbcd0385d558af6c74571aaac9020d7cbb9e56cc4780299bd420559f29bb1
+distfiles="https://github.com/cilynx/rtl88x2bu/archive/${_gitrev}.tar.gz"
+checksum=d1ed87d40cb47c1ac011a9e57aba84d8efecead84f0ddab4ea2847281c2749ca
 dkms_modules="88x2bu ${version}"
 
 case "$XBPS_TARGET_MACHINE" in

